pkgs/applications/video/mkvtoolnix/default.nix

```diff
@@ -25,7 +25,7 @@ stdenv.mkDerivation rec {
     expat
     file xdg_utils boost libebml zlib
     libmatroska libogg libvorbis flac
-    (optional withGUI qt5.qtbase)
+    (optional withGUI qt5.qtbase.dev)
   ];
```

Comment:
It is not clear to me yet why the derivation works with
```
  buildInputs = [
    expat file xdg_utils boost libebml zlib libmatroska libogg
    libvorbis flac
    (optional withGUI qt5.qtbase.dev)
  ];
```
as well as with
```
  buildInputs = [
    expat file xdg_utils boost libebml zlib libmatroska libogg
    libvorbis flac
  ] ++ optional withGUI qt5.qtbase;
```
but does not work with
```
  buildInputs = [
    expat file xdg_utils boost libebml zlib libmatroska libogg
    libvorbis flac
    (optional withGUI qt5.qtbase)
  ];
```
As explained by @dezgeg and @rnhmjoj I think it should not work with the first solution either.

Anyway, the nix expression is fixed now and works for me.